src/bedrock/api.js

```diff
@@ -94,8 +94,122 @@ module.exports = class BedrockRealmAPI extends RealmAPI {
     return await this.rest.put(`/worlds/${realmId}/${state}`)
   }
 
+  async deleteRealm (realmId) {
+    return await this.rest.delete(`/worlds/${realmId}`)
+  }
+
   async getRealmWorldDownload (realmId, slotId, backupId = 'latest') {
     const data = await this.rest.get(`/archive/download/world/${realmId}/${slotId}/${backupId}`) // if backupId = latest will get the world as it is now not the most recent backup
     return new Download(this, data)
   }
+
+  async getRealmSubscriptionInfo (realmId) {
+    return await this.rest.get(`/subscriptions/${realmId}`)
+  }
+
+  async getRealmSubscriptionInfoDetailed (realmId) {
+    return await this.rest.get(`/subscriptions/${realmId}/details`)
+  }
+
+  async changeRealmActiveSlot (realmId, slotId) {
+    return await this.rest.put(`/worlds/${realmId}/slot/${slotId}`)
+  }
+
+  async changeRealmNameAndDescription (realmId, name, description) {
+    return await this.rest.put(`/worlds/${realmId}`, {
+      body: {
+        name,
+        description
+      }
+    })
+  }
+
+  async resetRealm (realmId) {
+    return await this.rest.put(`/worlds/${realmId}/reset`)
+  }
+
+  // Reference https://github.com/PrismarineJS/prismarine-realms/issues/34 for configuration structure
+  async changeRealmConfiguration (realmId, configuration) {
+    return await this.rest.put(`/worlds/${realmId}/configuration`, {
+      body: configuration
+    })
+  }
+
+  async removeRealmInvite (realmId, uuid) {
+    const data = await this.rest.put(`/invites/${realmId}/invite/update`, {
+      body: {
+        invites: {
+          [uuid]: 'REMOVE'
+        }
+      }
+    })
+    return new Realm(this, data)
+  }
+
+  async opRealmPlayer (realmId, uuid) {
+    return await this.rest.put(`/invites/${realmId}/invite/update`, {
+      body: {
+        invites: {
+          [uuid]: 'OP'
+        }
+      }
+    })
+  }
+
+  async deopRealmPlayer (realmId, uuid) {
+    return await this.rest.put(`/invites/${realmId}/invite/update`, {
+      body: {
+        invites: {
+          [uuid]: 'DEOP'
+        }
+      }
+    })
+  }
+
+  async removeAllRealmPlayers (realmId) {
+    const data = await this.rest.put(`/invites/${realmId}/invite/update`, {
+      body: {
+        invites: null
+      }
+    })
+    return new Realm(this, data)
+  }
+
+  async banPlayerFromRealm (realmId, uuid) {
+    return await this.rest.post(`/worlds/${realmId}/blocklist/${uuid}`)
+  }
+
+  async unbanPlayerFromRealm (realmId, uuid) {
+    return await this.rest.delete(`/worlds/${realmId}/blocklist/${uuid}`)
+  }
+
+  async removeRealmFromJoinedList (realmId) {
+    return await this.rest.delete(`/invites/${realmId}`)
+  }
+
+  async changeForcedTexturePack (realmId, forced) {
```

Comment:
changeForcedTexturePack here sounds clearer to me as "changeIsTexturePackRequired"